HW4_Sapozhnikov/prototool.py

```diff
@@ -0,0 +1,564 @@
+    # Form an output dictionary
+    alignment_dict['aligned_seq_on'] = aligned_seq_on
+
+    identity = round(counter_identity/alignment_length, 4)
+
+    alignment_dict[f'aligned_seq{seq_id+1}'] = {'seq': aligned_seq2,
+                                                'length': alignment_length,
+                                                'score': alignment_score,
+                                                'identity': identity,
+                                                'gaps': counter_gaps}
+
+    # Prettify an alignment output
+    if prettify is True:
+        prettify_alignment(aligned_seq_on, aligned_seq2)
+    else:
+        pass
+
+    return alignment_dict
+
+
+def count_protein_molecular_weight(*seqs_list: Union[List[str], str]) -> dict:
+    """
+    :param seqs_list: seqs_list is a list of strings without whitespace
+    (e.g. 'AlaSer'). You can put as many sequences as you wish.
+    :return: This function returns molecular weight of the protein.
+    """
+    result = {}
+    for seq in seqs_list:
+        protein_weight = 0
+        aminoacids = [seq[i:i + 3] for i in range(0, len(seq), 3)]
+        for i, aminoacid in enumerate(aminoacids):
+            if aminoacid in AMINOACIDS_DICT.keys():
+                aminoacid_weight = (AMINOACIDS_DICT[aminoacid]
+                                    ['MOLECULAR_WEIGHTS'])
+                protein_weight += aminoacid_weight
+                result[seq] = protein_weight
+    return result
+
+
+def from_proteins_seqs_to_rna(*seqs_list: Union[List[str], str]) -> dict:
+    """
+    :param seqs_list: a list of strings with type 'ValTyrAla','AsnAspCys'.
+    You can pass more than one sequence at the time.
+    :return: dictionary, where [key] is your input protein sequences
+    and values are combinations of RNA codones, which encode this protein
+    """
+
+    answer_dictionary = {}
+    for seq in seqs_list:
+        
+        rna_combination = ''
+        divided_acids = [seq[i:i + 3] for i in range(0,
+                                                     len(seq),
+                                                     3)]
+        for divided_acid in divided_acids:
+            
+            if divided_acid in AMINOACIDS_DICT.keys():
+                rna_combination += next(iter(AMINOACIDS_DICT[divided_acid]
+                                             ['PROTEIN_TO_RNA_COMBINATION']))
+            else:
+                raise ValueError('Non-protein aminoacids in sequence')
+        answer_dictionary[seq] = rna_combination
+    return answer_dictionary
+
+
+def isoelectric_point_determination(*seqs_list: Union[List[str], str]) -> dict:
+    """
+    :param seqs_list: a list of strings with type 'ValTyrAla','AsnAspCys'.
+    You can pass more than one sequence at a time.
+    :return: dictionary, where [key] is your input protein sequence and value
+    is an isoelectric point of your input proteins
+    """
+    answer_dictionary = {}
+
+    for aminoacids in seqs_list:
```

Comment:
Мб тут и везде лучше не `seqs_list` (кстати слово лист тут вообще надо убрать), а `proteins`. Все таки на `seqs` в первую очередь думаешь про ДНК/РНК.
```suggestion
    for protein in proteins:
```